Use same styling for statuses in email as on web

c960657 opened the pull request:
The notification emails containing statuses are rendered with a different styling than on the web.

- White-space are collapsed, so ASCII art or spaces used for indentation will not work.
- There is no margin between paragraphs, so two consecutive newlines look like a single newline.
- Links are not displayed in their compact format (leading `https://www` is stripped and long path is truncated with ellipsis).
- A very long word will make the container grow horizontally (in Mail.app on macOS).

This PR does address the difference in font family and size. I assume that is intentional.

c960657 committed:
Use same CSS for status in email as on web

ClearlyClaire reviewed (approved):
Looks ok at a glance but I'd like another set of eyes, especially since both the design and its implementation have been carried out by people external from the core team with a lot of expertise in the field.

renchap commented:
Pinging @hteumeuleu if you can have a look, as you wrote this original CSS?

hteumeuleu commented:
Just had a quick look and this should be fine.

mjankowski commented:
I think this might have had unintended consequences on the welcome mailer...

Currently in main:

<img width="891" alt="Screenshot 2024-09-27 at 13 16 55" src="https://github.com/user-attachments/assets/ade26cb5-784c-4421-b92a-b6e0d206b881">

If I comment out the `table-layout` change from this PR:

<img width="879" alt="Screenshot 2024-09-27 at 13 17 04" src="https://github.com/user-attachments/assets/5dcd9a29-ae83-4063-b72f-485aa832b790">

c960657 commented:
Fix coming up (after dinner).

c960657 commented:
Follow-up PR created: #32132

@mjankowski I have fixed the issue with the collapsed banner image. Did you notice other problems? I might have missed something while comparing the two screenshots.

The text wrapping in the buttons look ugly, so I fixed that as well, even though this is not a new problem. On my machine, the font is a bit more compact than that in your screenshot, so here the “Personalize” text fits within the default button width.

mjankowski commented:
> Follow-up PR created: #32132
> 
> @mjankowski I have fixed the issue with the collapsed banner image. Did you notice other problems? I might have missed something while comparing the two screenshots.

That's all I noticed -- the vertical white line and missing instance card ... although I wasn't exhaustive across all mailers, just noticed that in welcome while looking at something unrelated.

> The text wrapping in the buttons look ugly, so I fixed that as well, even though this is not a new problem. On my machine, the font is a bit more compact than that in your screenshot, so here the “Personalize” text fits within the default button width.

🥳
